SurfaceRegistration/SurfaceRegistration.py

```diff
@@ -1242,25 +1250,74 @@ def displayROI(self, inputModelNode, scalarName):
         displayNode.SetScalarVisibility(True)
         displayNode.EndModify(disabledModify)
 
+    def getCellPoints(self, polyData, cellIdx):
+        points = vtk.vtkIdList()
+        polyData.GetCellPoints(cellIdx, points)
+        for i in range(points.GetNumberOfIds()):
+            yield points.GetId(i)
+
+    def getPointCells(self, polyData, pointIdx):
+        cells = vtk.vtkIdList()
+        polyData.GetPointCells(pointIdx, cells)
+        for i in range(cells.GetNumberOfIds()):
+            yield cells.GetId(i)
+
     def findROI(self, fidList):
         hardenModel = slicer.app.mrmlScene().GetNodeByID(fidList.GetAttribute("hardenModelID"))
         connectedModel = slicer.app.mrmlScene().GetNodeByID(fidList.GetAttribute("connectedModelID"))
         landmarkDescription = self.decodeJSON(fidList.GetAttribute("landmarkDescription"))
         arrayName = fidList.GetAttribute("arrayName")
-        ROIPointListID = vtk.vtkIdList()
-        for key,activeLandmarkState in landmarkDescription.items():
-            tempROIPointListID = vtk.vtkIdList()
-            if activeLandmarkState["ROIradius"] != 0:
-                self.defineNeighbor(tempROIPointListID,
-                                    hardenModel.GetPolyData(),
-                                    activeLandmarkState["projection"]["closestPointIndex"],
-                                    activeLandmarkState["ROIradius"])
-            for j in range(0, tempROIPointListID.GetNumberOfIds()):
-                ROIPointListID.InsertUniqueId(tempROIPointListID.GetId(j))
-        listID = ROIPointListID
-        self.addArrayFromIdList(listID, connectedModel, arrayName)
+
+        # Each ROI radius is a "weight" in the graph of points. Do a breadth-first
+        # traversal, subtracting one from the weight at each layer. Continue until
+        # weight is 0.
+
+        # Starting weights (the ROI radius at each fiducial)
+        weights = defaultdict(set)
+        for key, state in landmarkDescription.items():

```

Comment:
`for key, state in landmarkDescription.items()` -> `for state in landmarkDescription.values()`

Reply:
This is done.